app/main.py

```diff
@@ -0,0 +1,40 @@
+import subprocess
+import time
+from contextlib import asynccontextmanager
+
+from fastapi import FastAPI
+from starlette.middleware.cors import CORSMiddleware
+from api.api_v1.api import api_router
+from core.config import settings
+from fastapi.middleware.httpsredirect import HTTPSRedirectMiddleware
+import os
+from init.db_init import init_db
+
+@asynccontextmanager
+async def lifespan(app: FastAPI):
+    # Sleep necessary because the db might not be up before this call, which results in a crash (restart the backend container for it to work properly though)
+
+    # auto migrate the db
+    time.sleep(3)
+    subprocess.run(["alembic", "revision", "--autogenerate"], check=True)
+    subprocess.run(["alembic", "upgrade", "head"], check=True)
+    time.sleep(1)
+    init_db()
+
+    yield
+
+app = FastAPI(
+    title="OpenAgri", openapi_url="/api/v1/openapi.json", lifespan=lifespan
```

Comment:
title shoud `Pest and Desease` instead of `OpenAgri` right?

Reply:
Yes, will change to PDM